core/src/main/java/org/lflang/analyses/statespace/Tag.java

```diff
@@ -0,0 +1,65 @@
+/**
+ * Class representing a logical time tag, which is a pair that consists of a timestamp (type long)
+ * and a microstep (type long).
+ *
+ * @author{Shaokai Lin <shaokai@berkeley.edu>}
+ */
+package org.lflang.analyses.statespace;
+
+import org.lflang.TimeValue;
+
+public class Tag implements Comparable<Tag> {
```

Comment:
This class looks like it could be a record
```java
public record Tag(long timestamp, long microstep, boolean forever)
   implements Comparable<Tag> { // ...
```

This would spare you the fields and ctor declaration, and the equals declaration.